gluoncv/model_zoo/rpn/rpn.py

```diff
@@ -52,54 +53,175 @@ class RPN(gluon.HybridBlock):
         Return top proposal results after NMS in testing.
     min_size : int
         Proposals whose size is smaller than ``min_size`` will be discarded.
+        multi_level : boolean
+        Whether to extract feature from multiple level. This is used in FPN.
 
     """
-    def __init__(self, channels, stride, base_size, scales, ratios, alloc_size,
+
+    def __init__(self, channels, strides, base_size, scales, ratios, alloc_size,
                  clip, nms_thresh, train_pre_nms, train_post_nms,
-                 test_pre_nms, test_post_nms, min_size, **kwargs):
+                 test_pre_nms, test_post_nms, min_size, multi_level=False, **kwargs):
         super(RPN, self).__init__(**kwargs)
-        weight_initializer = mx.init.Normal(0.01)
+        self._nms_thresh = nms_thresh
+        self._multi_level = multi_level
+        self._train_pre_nms = max(1, train_pre_nms)
+        self._train_post_nms = max(1, train_post_nms)
+        self._test_pre_nms = max(1, test_pre_nms)
+        self._test_post_nms = max(1, test_post_nms)
+        num_stages = len(scales)
         with self.name_scope():
-            self.anchor_generator = RPNAnchorGenerator(
-                stride, base_size, ratios, scales, alloc_size)
-            anchor_depth = self.anchor_generator.num_depth
-            self.region_proposaler = RPNProposal(
-                clip, nms_thresh, train_pre_nms, train_post_nms,
-                test_pre_nms, test_post_nms, min_size, stds=(1., 1., 1., 1.))
-            self.conv1 = nn.HybridSequential()
-            self.conv1.add(nn.Conv2D(channels, 3, 1, 1, weight_initializer=weight_initializer))
-            self.conv1.add(nn.Activation('relu'))
-            # use sigmoid instead of softmax, reduce channel numbers
-            self.score = nn.Conv2D(anchor_depth, 1, 1, 0, weight_initializer=weight_initializer)
-            self.loc = nn.Conv2D(anchor_depth * 4, 1, 1, 0, weight_initializer=weight_initializer)
+            if self._multi_level:
+                asz = alloc_size
+                self.anchor_generator = nn.HybridSequential()
+                for _, st, s in zip(range(num_stages), strides, scales):
```

Comment:
hey, I think there may be a little mistake?
every stage_anchor_generator should be initialized with scales=(2, 4, 8, 16, 32), not just one of them.
I ran your codes to train mask_rcnn with fpn, but got very huge loss.
After I changed the scales problem, the loss looks common now.

by the way, is the pretrained model of mask_rcnn_fpn_resnet50_v1b_coco available now?
 I got a mistake when loading the params you provided, which is 
'Parameter 'box_featueres.1.weight' is missing in file './models/mask_rcnn_fpn_resnet50_v1b_coco-1364d0af.params', which contains parameters: 'features.resnetv1b_conv0_weight', 'features.resnetv1b_batchnorm0_gamma', 'features.resnetv1b_batchnorm0_beta', ..., 'mask.deconv.8.weight', 'mask.deconv.8.bias', 'mask.mask.weight', 'mask.mask.bias'. Set allow_missing=True to ignore missing parameters.'